File: ExpandableListSimple/app/src/main/java/com/expandablelistsimple/expandablelist/SimpleExpandableListView.java
package com.expandablelistsimple.expandablelist;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ExpandableListView;

/**
 * by y on 2016/11/21
 */

public class SimpleExpandableListView extends ExpandableListView {

    public SimpleExpandableListView(Context context) {
        super(context);
    }

    public SimpleExpandableListView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SimpleExpandableListView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }


    @Override
    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int expandSpec = MeasureSpec.makeMeasureSpec(Integer.MAX_VALUE >> 2, MeasureSpec.AT_MOST);
        super.onMeasure(widthMeasureSpec, expandSpec);
    }
}

File: utils_expandablelist/src/main/java/sample/util/develop/android/expandablelist/widget/SimpleExpandableListView.kt
package sample.util.develop.android.expandablelist.widget

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.ExpandableListView

class SimpleExpandableListView : ExpandableListView {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )


    public override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val expandSpec =
            View.MeasureSpec.makeMeasureSpec(Integer.MAX_VALUE shr 2, View.MeasureSpec.AT_MOST)
        super.onMeasure(widthMeasureSpec, expandSpec)
    }
}
